crates/ere-risc0/src/compile_stock_rust.rs

```diff
@@ -0,0 +1,123 @@
+use crate::compile::Risc0Program;
+use crate::error::CompileError;
+use cargo_metadata::MetadataCommand;
+use risc0_binfmt::ProgramBinary;
+use std::fs;
+use std::path::Path;
+use std::process::Command;
+use tracing::info;
+
+static CARGO_ENCODED_RUSTFLAGS_SEPARATOR: &str = "\x1f";
+// TODO: Make this with `zkos` package building to avoid binary file storing in repo.
+// File taken from https://github.com/risc0/risc0/blob/v3.0.1/risc0/zkos/v1compat/elfs/v1compat.elf
```

Comment:
what is this?

Reply:
This is "kernel elf" binary which has to be used together with guest program binary elf to make proper risc0 program which can be executed on this zkvm. Using risc0 sdk it was previously done with ` risc0_build::build_package`

Reply:
Ah right, could we make an issue with the TODO above? And also an issue to have separate Dockerfiles for the stock compilers. These aren't blocking right now or urgent

main thing with the elf file is that if it becomes out of sync

Reply:
https://linear.app/zkevm/issue/ZKE-294/build-kernel-elf-for-risc0-instead-of-storing-it-in-the-ere-repo
https://linear.app/zkevm/issue/ZKE-295/create-separated-dockerfiles-for-guest-program-build-with-rust-stock